templates/contest/list.html

```diff
@@ -132,7 +132,7 @@ <h4>{{ _('Ongoing Contests') }}</h4>
                             </div>
                         </td>
                         <td>
-                            {% if contest.show_scoreboard or perms.judge.see_private_contest %}
+                            {% if contest.show_scoreboard or contest.can_see_scoreboard(request) %}
```

Comment:
I feel like the performance of this could be really bad when contest pages are unpaginated and a lot of contests with hidden scoreboards show up. But I am going to let you merge this now and deal with it later.